src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,205 @@
+  onClose,
+  closeButtonText,
+  position,
+  ...other
+}) => {
+  /**
+   * Initialize states, constants, and refs
+   */
+  const [isMounted, setIsMounted] = useState(false);
+  const [activeFocus, setActiveFocus] = useState(isActive || false);
+  const ref = useRef<HTMLElement | null>(null);
+  /**
+   * Get previous prop
+   * 1) This is used to compare the previous prop to the current prop
+   */
+  function usePrevious(isActive: any) {
+    useEffect(() => {
+      ref.current = isActive;
+    });
+    return ref.current;
+  }
+
+  /**
+   * Use effect
+   * 1) Set prevIsActive to previous isActive prop
+   * 2) If prevIsActive is defined and previous isActive prop is not equal
+   * to current isActive prop, toggle state
+   */
+  const prevIsActive = usePrevious(isActive); /* 1 */
+  useEffect(() => {
+    if (prevIsActive !== undefined || null) {
+      if (isActive) {
+        activateDOM();
+      } else {
+        deactivateDOM();
+      }
+    }
+  });
+
+  /**
+   * Set a flag when this item has mounted
+   * Otherwise, portals will render which are not handled well
+   * between React and Next.js
+   */
+  useEffect(() => {
+    setIsMounted(true);
+  }, [setIsMounted]);
+
+  /**
+   * Activate DOM
+   * 1) Open the Popover
+   * 2) Set activeFocus state to true
+   * 3) This accommodates popover animation so that it auto receives focus
+   */
+  function activateDOM() {
+    /* 3 */
+    setTimeout(() => {
+      setActiveFocus(true); /* 2 */
+    }, 300);
+  }
+
+  /**
+   * Deactivate DOM
+   * 1) Close the popover
+   * 2) Set activeFocus state to false
+   */
+  function deactivateDOM() {
+    setActiveFocus(false); /* 2 */
+  }
+
+  /**
+   * Handle onClose
+   * 1) Close the popover
+   * 2) Run the onClose prop (pass in function) if it exists
+   */
+  function handleOnClose(e: KeyboardEvent | MouseEvent) {
+    deactivateDOM(); /* 1 */
+
+    if (onClose) {
+      onClose(e); /* 2 */
+    }
+  }
+
+  /**
+   * Handle onKeyDown
+   * 1) If escape button is struck, close the popover
+   */
+  const handleOnKeyDown = (e: KeyboardEvent) => {
+    console.log(e.code);
+    if (e.code === ESCAPE_KEYCODE) {
+      handleOnClose(e); /* 1 */
+    }
+  };
+
+  const popoverHeader = oneByType(children, PopoverHeader);
+  const header = React.Children.map(popoverHeader, (child) => {
+    return React.cloneElement(child, {
+      onClick: (e) => handleOnClose(e),
+      dismissible: dismissible,
+    });
+  });
+  const body = oneByType(children, PopoverBody);
+  const footer = oneByType(children, PopoverFooter);
+
+  const componentClassName = clsx(
+    styles['popover'],
+    className,
+    isActive && [styles['eds-is-active']],
+    position === 'top-left' && [styles['popover--top-left']],
+    position === 'bottom-left' && [styles['popover--bottom-left']],
+    position === 'bottom-right' && [styles['popover--bottom-right']],
+  );
+
+  if (!isMounted) return null;
+
+  return (
+    <FocusLock disabled={!activeFocus}>
```

Comment:
oh wow complicated stuff, nice job! Did I hear correctly that there are more accessibility fixes/updates you wanted to make?

For reference, the Popover currently used in the Learning Platform is built on top of [HeadlessUI's Popover ](https://headlessui.dev/react/popover). Here's the component in traject: https://github.com/FB-PLP/traject/blob/master/app/assets/javascripts/v2/core/EDSCandidates/Popover/Popover.tsx

I know we'd prefer native components when possible, but Headless does have solid a11y support, and can be a great timesaver (it comes fully unstyled and only provides functionality)